### PR TITLE
Recognize Windows/worktree Git hooks and register recommend/team commands; improve runtime-health checks

### PR DESCRIPTION
**Pull request by harshanandak**
### Motivation

- Fix false runtime-prerequisite failures on Windows worktrees where `HOOKS_NOT_ACTIVE` and `LEFTHOOK_MISSING` were emitted despite valid hooks being installed.
- Reduce noise from the command registry by ensuring `recommend` and `team` export standard command metadata so loader warnings are not raised.
- Improve diagnostics and resilience of runtime-health checks so failures indicate the exact verification issue and preserve hard-stops only for real problems.

### Description

- Implement a layered hook-health verification in `lib/runtime-health.js` that preserves the `core.hooksPath` check and adds a `git rev-parse --git-path hooks` fallback, verifies presence of required hook files (`pre-commit`, `pre-push`), and returns richer verification states and messages. 
- Harden Lefthook detection logic integration so `missing-binary` does not become a hard-stop when effective hooks are present, and surface clearer `LEFTHOOK_MISSING` diagnostics when appropriate. 
- Add command metadata wrappers for `lib/commands/recommend.js` and `lib/commands/team.js` so they export `{ name, description, handler }` and are properly loaded by the registry. 
- Add test coverage and planning docs: new/updated tests in `test/runtime-health.test.js` and `test/forge-cli-registry.test.js` to cover Windows/worktree hook fallbacks, Lefthook edge-cases, and registry warnings; plus plan/design/task/decision docs under `docs/plans/2026-04-25-windows-worktree-hooks-health-*`. 

### Testing

- Ran unit tests covering runtime health checks with the updated `test/runtime-health.test.js` scenarios (worktree fallback active hooks, lefthook missing-binary vs missing-dependency behavior) and they passed. 
- Ran registry integration test `test/forge-cli-registry.test.js` which validates that `recommend` and `team` load without skip warnings and it passed. 
- Executed the full test suite for the modified modules and all related tests succeeded.

------
[Codex Task](https://chatgpt.com/codex/cloud/tasks/task_e_69ed238c8f1c8325a4e28f5e421daa57)

### CODE REVIEW

File: lib/runtime-health.js [hunk 1]
```diff
@@ -219,16 +268,19 @@ function checkRuntimeHealth(projectRoot, options = {}) {
...
   const bd = checkCommandAvailability('bd', root, options);
   const gh = checkCommandAvailability('gh', root, options);
   const jq = checkCommandAvailability('jq', root, options);
   const shell = normalizeShellRuntime(options.shellRuntime, platform, options);
 
   const diagnostics = [];
 
-  if (lefthook.state !== 'installed') {
+  const lefthookMissing = lefthook.state === 'missing-dependency'
+    || (lefthook.state === 'missing-binary' && !hooks.active);
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/runtime-health.js]: Fixed in commit 1a82d0d: restored LEFTHOOK_MISSING as a hard-stop whenever the declared Lefthook binary is missing, even if hook files are active. Updated the runtime-health regression test to cover this case.

File: docs/plans/2026-04-25-windows-worktree-hooks-health-tasks.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+5. **Commit:** `fix: improve lefthook detection for worktree environments`
+
+**Expected output:** Runtime health only reports `LEFTHOOK_MISSING` for real missing installs, not valid worktree environments.
+
+---
+
+## Task 4: Remove command-registry warning noise for recommend/team
+
+**File(s):** `lib/commands/recommend.js`, `lib/commands/team.js`, `test/commands/registry.test.js` (or nearest registry coverage file)  
+**OWNS:** `lib/commands/recommend.js`, `lib/commands/team.js`, `test/commands/registry.test.js`
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply docs/plans/2026-04-25-windows-worktree-hooks-health-tasks.md]: Fixed in commit 1a82d0d: corrected Task 4 to reference the actual registry coverage file, test/commands/_registry.test.js, in both File(s) and OWNS.

File: lib/runtime-health.js [hunk 3]
```diff
@@ -77,30 +78,78 @@ function checkHookInstallation(projectRoot, options = {}) {
...
-      active: false,
-      state: 'unverified',
-      hooksPath: null,
-      message: 'Git hooks could not be verified.'
+      hooksDir,
+      missingHooks,
+      message: active
+        ? ''
+        : `Git hooks directory is missing required hooks: ${missingHooks.join(', ')}.`
     };
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/runtime-health.js:179]: Fixed in commit 1649ab7: an explicit non-Lefthook core.hooksPath now returns inactive immediately instead of falling back to .git/hooks. Added regression coverage for the explicit custom hooksPath case.

File: lib/runtime-health.js [hunk 4]
```diff
@@ -77,30 +78,78 @@ function checkHookInstallation(projectRoot, options = {}) {
...
+      return path.isAbsolute(rawPath) ? rawPath : path.resolve(projectRoot, rawPath);
+    } catch {
+      return null;
+    }
+  }
+
+  function fileLooksLikeActiveHook(filePath) {
+    try {
+      const stat = fs.statSync(filePath);
+      return stat.isFile() && stat.size > 0;
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/runtime-health.js]: Fixed in commit 972c480: worktree fallback now requires hook files to invoke Lefthook before considering them active. Added regression coverage for non-Lefthook pre-commit/pre-push hook files.

File: test/runtime-health.test.js [hunk 5]
```diff
@@ -259,6 +263,71 @@ describe('runtime health checks', () => {
...
+
+    expect(result.hardStop).toBe(true);
+    expect(result.diagnostics).toContainEqual(
+      expect.objectContaining({
+        code: 'LEFTHOOK_MISSING',
+        severity: 'hard-stop'
+      })
+    );
+    expect(result.checks.lefthook.state).toBe('missing-binary');
+  });
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply test/runtime-health.test.js:426]: Resolved as not applied: this conflicts with the earlier P1 review requirement to preserve LEFTHOOK_MISSING as a hard-stop when the declared Lefthook binary is absent. The PR objective is to avoid false positives for valid Lefthook installations, not to allow a declared-but-missing Lefthook binary through stage entry.

File: lib/runtime-health.js [hunk 6]
```diff
@@ -77,30 +78,89 @@ function checkHookInstallation(projectRoot, options = {}) {
...
 
+    if (active) {
+      return {
+        active,
+        state: 'active',
+        verification: 'core.hooksPath',
+        hooksPath: hooksPath || null,
+        message: ''
+      };
+    }
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/runtime-health.js:150]: Fixed in commit a57149d: core.hooksPath matches now verify the configured .lefthook/hooks pre-commit and pre-push entries before reporting active. Added regression coverage for missing configured Lefthook hook entries.

File: lib/runtime-health.js [hunk 7]
```diff
@@ -77,30 +78,89 @@ function checkHookInstallation(projectRoot, options = {}) {
...
+    } catch {
+      return null;
+    }
+  }
+
+  function fileLooksLikeActiveHook(filePath) {
+    try {
+      const stat = fs.statSync(filePath);
+      if (!stat.isFile() || stat.size === 0) return false;
+      return /\blefthook\b/i.test(fs.readFileSync(filePath, 'utf8'));
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/runtime-health.js:119]: Fixed in commit e822674: fallback hook validation now requires POSIX execute permission in addition to file/content checks. Added regression coverage for non-executable Lefthook hook files.

File: lib/runtime-health.js [hunk 8]
```diff
@@ -77,30 +78,96 @@ function checkHookInstallation(projectRoot, options = {}) {
...
     });
 
-    const hooksPath = normalizeHooksPath(output, platform);
+    hooksPath = normalizeHooksPath(output, platform);
     const active = hooksPath === expectedRelativeHooksPath || hooksPath === expectedAbsoluteHooksPath;
 
+    if (active) {
+      const hooksDir = path.join(projectRoot, '.lefthook', 'hooks');
+      const missingHooks = requiredHooks.filter(hook => !fileLooksLikeActiveHook(path.join(hooksDir, hook)));
+      const verifiedActive = missingHooks.length === 0;
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/runtime-health.js]: Fixed in commit e822674: relative core.hooksPath validation now resolves against git rev-parse --show-toplevel, so nested invocations check the repository root .lefthook/hooks directory. Added regression coverage for nested projectRoot execution.

File: lib/runtime-health.js [hunk 9]
```diff
@@ -77,30 +78,96 @@ function checkHookInstallation(projectRoot, options = {}) {
...
+
+  function fileLooksLikeActiveHook(filePath) {
+    try {
+      const stat = fs.statSync(filePath);
+      if (!stat.isFile() || stat.size === 0) return false;
+      return /\blefthook\b/i.test(fs.readFileSync(filePath, 'utf8'));
+    } catch {
+      return false;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (harshanandak) [reply lib/runtime-health.js:123]: Fixed in commit e822674: POSIX hook detection now checks execute permission before accepting hook files as active, and the test fixture marks generated Lefthook hooks executable. Added explicit non-executable hook coverage.
